fraction.rb

```diff
@@ -1,64 +1,51 @@
 class Fraction
   def initialize(numerator, denominator)
-    if ! numerator.is_a?(Integer) && denominator.is_a?(Integer)
-      # TODO: Proper execption for this issue
-      # TODO: Handle case when one or both parameters are not int but could be easily transformed to int
-      raise 'these are supposed to be numbers: '+numerator.inspect+' '+ denominator.inspect
-    end
     @numerator = numerator
     @denominator = denominator
   end
-
   def numerator
     @numerator
   end
-
   def denominator
     @denominator
   end
-
-  def simplify
-
-=begin
-     if @numerator > @denominator then b=@denominator else b=@numerator
+  def simple
+    a = @numerator
+    b = @denominator
+    while (a != 0) && (b != 0)
+      if a > b
+        a = a % b
+      else
+        b = b % a
+      end
     end
-    end
-
-    while b>1
-     if (@numerator%b == 0 and @denominator%b == 0)
-       @denominator = @denominator /b
-       @numerator = @numerator /b
-     end
-      b=b-1
+    devider = a + b
+=begin
+    @numerator = @numerator/devider    # вот это другое дело!
+    @denominator = @denominator/devider
 =end
-    Fraction.new(@numerator / @numerator.gcd(@denominator), @denominator / @denominator.gcd(@numerator))
+    Fraction.new(@numerator = @numerator/devider, @denominator = @denominator/devider) #непрозрачный способ!!!
```

Comment:
Do you want to create new `Fraction` object or modify existing one? You do both, it is not right.